R/derive_param_wbc_abs.R

```diff
@@ -0,0 +1,172 @@
+#' Add a parameter for lab differentials converted to absolute values
+#'
+#' @description
+#'
+#' Add a parameter by converting lab differentials from percentage to absolute values
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `by_vars` argument, `PARAMCD`, and `AVAL`
+#'   are expected to be present.
+#'
+#'   The variable specified by `by_vars` and `PARAMCD` must be a unique key of
+#'   the input dataset, and to the parameters specified by `wbc_code` and `diff_code`.
+#'
+#' @param set_values_to Variables to set
+#'
+#'   A named list returned by `vars()` defining the variables to be set for the
+#'   new parameter, e.g. `vars(PARAMCD = "LYMPH", PARAM = "Lymphocytes Abs (10^9/L)")` is
+#'   expected.
+#'
+#' @param get_unit_expr An expression providing the unit of the parameter
+#'
+#'   The result is used to check the units of the input parameters.
+#'
+#'   Permitted Values: a variable containing unit from the input dataset, or a function call,
+#'   for example, `get_unit_expr = extract_unit(PARAM)`.
+#'
+#' @param wbc_unit A string containing the required unit of the WBC parameter
+#'
+#'   Default: `"10^9/L"`
+#'
+#' @param diff_type A string specifying the type of differential
+#'
+#'   Permitted Values: `"percent"`, `"fraction"`
+#'
+#' @param by_vars Grouping variables
+#'
+#'   Permitted Values: list of variables
+#'
+#' @param wbc_code White Blood Cell (WBC) parameter
+#'
+#'   The observations where `PARAMCD` equals the specified value are considered
+#'   as the WBC absolute results to use for converting the differentials.
+#'
+#'   Default: `"WBC"`
+#'
+#'   Permitted Values: character value
+#'
+#' @param diff_code white blood differential parameter
+#'
+#'   The observations where `PARAMCD` equals the specified value are considered
+#'   as the white blood differential lab results in percentage value to be converted
```

Comment:
Should be "lab results in fraction or percentage value......"

Reply:
Updated.